src/send-and-sync.md

```diff
@@ -74,7 +74,140 @@ of their pervasive use of raw pointers to manage allocations and complex ownersh
 Similarly, most iterators into these collections are Send and Sync because they
 largely behave like an `&` or `&mut` into the collection.
 
+## Example
+
+[`Box`][box-doc] is implemented as it's own special intrinsic type by the
+compiler for [various reasons][box-is-special], but we can implement something
+with similar-ish behavior ourselves to see an example of when it is sound to
+implement Send and Sync. Let's call it a `Carton`.
+
+We start by writing code to take a value allocated on the stack and transfer it
+to the heap.
+
+```rust,ignore
+use std::mem::size_of;
+use std::ptr::NonNull;
```

Comment:
```suggestion
use std::ptr;
```

It is customary to use the functions within the `ptr` module with that `ptr` prefix: they read much better: consider `NonNull<usize>`, where it is not clear what is not non-zero, and `ptr::NonNull< usize >`. Similarly, `ptr::null_mut()` reads way better than `null_mut()` 🙂

Reply:
Fixed